src/frontend/src/eth/services/erc20.services.ts

```diff
@@ -83,3 +87,83 @@ export const loadErc20Contracts = async (): Promise<{ success: boolean }> => {
 
 	return { success: true };
 };
+
+export const saveErc20Contract = async ({
+	contractAddress,
+	metadata,
+	updateSaveProgressStep,
+	modalNext,
+	onSuccess,
+	onError
+}: {
+	contractAddress: string;
+	metadata: Erc20Metadata | undefined;
+	updateSaveProgressStep: (step: ProgressStepsAddToken) => void;
+	modalNext: () => void;
+	onSuccess: () => void;
+	onError: () => void;
+}): Promise<void> => {
+	const $i18n = get(i18n);
```

Comment:
For consistency reason, expect i18n, can you pass those state's values as variables of the function?

Reply:
on it